### PR TITLE
Fixing IPv6 in Connection Factory

### PR DESCRIPTION
**Pull request by Rooks103**
In reference to issues #828 and #1025.

### CODE REVIEW

**Comment by adejanovski** [comment]
Thanks heaps @Rooks103!! 🙏 

@cmvelo, do you have a way to test that this PR fixes your issue?

**Comment by Rooks103** [comment]
@adejanovski we are working on getting a build at the moment to test in our environments (I work with @cmvelo). Didn't realize that issue #828 already had a potential fix.

**Comment by adejanovski** [comment]
> @adejanovski we are working on getting a build at the moment to test in our environments (I work with @cmvelo). 

ooooh ok 😅 
Let me know how testing goes.

**Comment by adejanovski** [comment]
I think a unit test would be good to protect us from regressions if you could write one.

**Comment by cmvelo** [comment]
@adejanovski — doesn't look like the current form of the fix is helping — seeing the following when trying add an IPv6 cluster with the new jar:

```
INFO   [dw-71 - POST /cluster/auth] i.c.r.ClusterResource - POST addOrUpdateCluster called with seedHost: 2600:1f18:65cc:ec0e:90bf:3127:7686:c1c3
WARN   [dw-71 - POST /cluster/auth] c.d.d.c.p.EC2MultiRegionAddressTranslator - Error resolving /2600:1f18:65cc:ec0e:90bf:3127:7686:c1c3, returning it as-is
javax.naming.NameNotFoundException: DNS name not found [response code 3]
```

---

We have the current settings in our `cassandra-reaper.yaml` file, since this reaper instance is also connected to a IPv4 multi-region cluster as well:
```
useAddressTranslator: true
jmxAddressTranslator:
  type: ec2MultiRegion
```

**Comment by cmvelo** [comment]
@adejanovski / @Rooks103 — with more investigation, it looks like the fix in [#828](https://github.com/thelastpickle/cassandra-reaper/issues/828) is for running Reaper in sidecar mode:

**Comment by Rooks103** [comment]
> I think a unit test would be good to protect us from regressions if you could write one.

@adejanovski I will get a unit test implemented once we get a fix that works in our environments (assuming we can).

File: src/server/src/test/java/io/cassandrareaper/jmx/JmxConnectionFactoryTest.java [hunk 1]
```diff
@@ -36,7 +40,7 @@
 public class JmxConnectionFactoryTest {
 
   @Test
-  public void fetchingJmxCredentialsPrioritizesThoseInSotrageFirst() {
+  public void fetchingJmxCredentialsPrioritizesThoseInStorageFirst() {
```
Comment (Rooks103) [inline src/server/src/test/java/io/cassandrareaper/jmx/JmxConnectionFactoryTest.java:43]: This was just a typo fix. Hopefully that's ok.

**Comment by Rooks103** [comment]
@adejanovski Unit Tests added. I moved that host code into its own method to make testing a bit simpler. 

We also did get a build of this code deployed and confirmed it works with IPv6 clusters and IPv4 clusters.

**Review by adejanovski — APPROVED** [review]
LGTM, thanks!

**Pull request merged**